### ISSUE TITLE
 About the full checkpoint of Mistral-E5

### ISSUE THREAD

**Issue by Iambestfeed** [microsoft/unilm#1430]
I had a passing idea about whether it is possible to use quantization for embedding models using Mistral?

However, one problem is that currently the checkpoint lacks the ``lm-head`` part, so I'm wondering if there would be any difference if using the full checkpoint for quantization instead of just the embedding part?

**Comment by intfloat** [comment]
@Iambestfeed Why would you want to include `lm_head` for quantization? Embedding models do not need it anyway.

**Comment by Iambestfeed** [comment]
@intfloat I'm looking at quantization algorithms like AWQ, GPTQ and they seem to work on minimizing loss based on the model's output (so I'm hoping to have ``lm_head`` so I can test to see if can use these algorithms?

**Comment by intfloat** [comment]
What about minimizing the MSE loss between the embedding vectors before and after quantization? Using `lm_head` makes little sense for embeddings, they are not fine-tuned with other parameters.

**Comment by Iambestfeed** [comment]
@intfloat Hmm, finetuning with MSE loss sounds like a practical idea. Do you think I should implement with fp16 model as a teacher and 4-bit model as a student model?

**Comment by intfloat** [comment]
If you want to optimize GPU memory usage and speed up inference, that surely makes sense.
